src/nnet3/nnet-cudnn-simple-component.cc

```diff
@@ -638,11 +638,11 @@ void CuDNN3DConvolutionComponent::Write(std::ostream &os, bool binary) const {
   KALDI_ASSERT(filter_dims[1] == input_num_filters_);
   KALDI_ASSERT(filter_dims[0] == num_filters_);
   WriteToken(os, binary, "<FilterXDim>");
-  WriteBasicType(os, binary, filter_dims[2]);
```

Comment:
I need  to verify whether this is correct. I'm afraid it may be, which means there are other parts of the code which may have the same problem.